### PR TITLE
Issue #DP-875: feat- Added implementation to handle composed h5p file

### PR DESCRIPTION
**Pull request by rheafernandes**

### CODE REVIEW

File: platform-modules/mimetype-manager/src/main/scala/org/sunbird/mimetype/mgr/impl/H5PMimeTypeMgrImpl.scala [hunk 1]
```diff
@@ -31,10 +37,13 @@ class H5PMimeTypeMgrImpl(implicit ss: StorageService) extends BaseMimeTypeManage
...
 
-    override def upload(objectId: String, node: Node, fileUrl: String, filePath: Option[String])(implicit ec: ExecutionContext): Future[Map[String, AnyRef]] = {
+    override def upload(objectId: String, node: Node, fileUrl: String, filePath: Option[String], params: UploadParams)(implicit ec: ExecutionContext): Future[Map[String, AnyRef]] = {
         validateUploadRequest(objectId, node, fileUrl)
         val file = copyURLToFile(objectId, fileUrl)
-        upload(objectId, node, file, filePath)
+        if (StringUtils.equalsIgnoreCase(params.fileFormat, COMPOSED_H5P_ZIP))
+            Future(Map[String, AnyRef]("identifier" -> objectId, "artifactUrl" -> fileUrl, "downloadUrl" -> fileUrl, "size" -> getFileSize(file).asInstanceOf[AnyRef]))
+        else
+            upload(objectId, node, file, filePath, params)
```
Comment (maheshkumargangula) [inline platform-modules/mimetype-manager/src/main/scala/org/sunbird/mimetype/mgr/impl/H5PMimeTypeMgrImpl.scala:46]:
```suggestion
            upload(objectId, node, file, filePath, params)
```
Use the upload method only. We should use the local storage for the content.

File: platform-modules/mimetype-manager/src/main/scala/org/sunbird/models/UploadParams.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package org.sunbird.models
+
+import play.api.mvc.QueryStringBindable
+
+
+case class UploadParams(fileFormat: String = "", validation: Boolean = true)
+
+object UploadParams {
```
Comment (maheshkumargangula) [inline platform-modules/mimetype-manager/src/main/scala/org/sunbird/models/UploadParams.scala:8]: Move this object to the content-service module.

File: platform-modules/mimetype-manager/pom.xml [hunk 3]
```diff
@@ -80,6 +84,17 @@
...
+            <groupId>com.typesafe.play</groupId>
+            <artifactId>play_${scala.major.version}</artifactId>
+            <version>${play2.version}</version>
+            <exclusions>
+                <exclusion>
+                    <artifactId>guava</artifactId>
+                    <groupId>com.google.guava</groupId>
+                </exclusion>
+            </exclusions>
+        </dependency>
```
Comment (maheshkumargangula) [inline platform-modules/mimetype-manager/pom.xml:97]: remove this dependency.

File: platform-modules/mimetype-manager/src/main/scala/org/sunbird/mimetype/mgr/impl/H5PMimeTypeMgrImpl.scala (hunk 1 above)
Comment (rheafernandes) [reply platform-modules/mimetype-manager/src/main/scala/org/sunbird/mimetype/mgr/impl/H5PMimeTypeMgrImpl.scala]: done

File: platform-modules/mimetype-manager/pom.xml (hunk 3 above)
Comment (rheafernandes) [reply platform-modules/mimetype-manager/pom.xml]: removed query string binder

File: platform-modules/mimetype-manager/src/main/scala/org/sunbird/models/UploadParams.scala (hunk 2 above)
Comment (rheafernandes) [reply platform-modules/mimetype-manager/src/main/scala/org/sunbird/models/UploadParams.scala]: removed query string binder

**Pull request merged**